crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -82,14 +83,105 @@ impl StaticFileProvider {
     }
 
     /// Creates a new [`StaticFileProvider`] with read-only access.
-    pub fn read_only(path: impl AsRef<Path>) -> ProviderResult<Self> {
-        Self::new(path, StaticFileAccess::RO)
+    ///
+    /// `watch_directory` will keep track of changes on the directory and re-initialize its inner
+    /// index accordingly (eg. node pushes a new block).
```

Comment:
can we improve the docs here a bit, these aren't super helpful  and it's not very clear why this is necessary or what this means.